activerecord/lib/active_record/persistence.rb

```diff
@@ -416,6 +416,8 @@ def toggle!(attribute)
     #   end
     #
     def reload(options = nil)
+      self.class.connection.clear_query_cache
```

Comment:
I don't think `uncached` is a good solution, as explained in https://github.com/rails/rails/pull/17140#issuecomment-57899597 and https://github.com/rails/rails/pull/17140#issuecomment-58480451.